src/components/ItemRecognition/ItemRecognition.js

```diff
@@ -75,19 +78,40 @@ class ItemRecognition extends Component {
     return (Date.now() - this.scanningStartTime) / 1000 > seconds;
   }
 
+  isItemRecognised = items => {
+    this.predictionStack.push(items.slice(0, ML_CERTAINTY));
+    const predictionCount = this.predictionStack.length;
+    if (predictionCount < ML_CERTAINTY) return false;
+    for (let i = 1; i < ML_CERTAINTY; i++) {
```

Comment:
Found this a bit hard to read initially, would inverting the logic and extracting to some booleans help?

How does something like this read in comparison?

```
isItemRecognised = items => {
  this.predictionStack.push(items.slice(0, ML_CERTAINTY))

  if (this.predictionsStack.length < ML_CERTAINTY) {
    return false
  }

  const predictionRatioThreshold = 1.2

  const predictions = this.predictionStack.slice(-ML_CERTAINTY) // Get only the number we care about. Needs a better name as it's really of the form [[prediction, prediction2, prediction3, ...], [...]] 
  
  const { id } = predictions[0][0]
  const areAllBestPredictionsTheSame = predictions.slice(1).every(([bestPrediction]) => bestPrediction.id === id)

  const areAllBestPredictionsCertainEnough = predictions.every(([bestPrediction, secondBestPrediction]) => bestPrediction.value / secondBestPrediction.value > predictionRatioThreshold)

  return areAlllBestPredictionsTheSame && areAllBestPredictionsCertainEnough
```

What do you think?

Reply:
Much easier to follow, thanks.